plugins/fuse/plugin_tests/server_fuse_test.py

```diff
@@ -0,0 +1,484 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import datetime
+import fuse
+import os
+import six
+import stat
+import tempfile
+import time
+
+from girder import config
+from girder.constants import AccessType
+from girder.models.file import File
+from girder.models.user import User
+from tests import base
+
+
+def setUpModule():
+    curConfig = config.getConfig()
+    curConfig.setdefault('server_fuse', {})
+    curConfig['server_fuse']['path'] = tempfile.mkdtemp()
+    base.enabledPlugins.append('fuse')
+    base.startServer()
+
+
+def tearDownModule():
+    curConfig = config.getConfig()
+    tempdir = curConfig['server_fuse']['path']
+    base.stopServer()
+    retries = 0
+    while retries < 50:
+        try:
+            os.rmdir(tempdir)
+            break
+        except OSError:
+            retries = 1
+            time.sleep(0.1)
+
+
+class ServerFuseTestCase(base.TestCase):
+    def setUp(self):
+        super(ServerFuseTestCase, self).setUp()
+        self.admin = User().findOne({'login': 'admin'})
+        self.user = User().findOne({'login': 'user'})
+        self.user2 = User().findOne({'login': 'second'})
+        curConfig = config.getConfig()
+        self.mainMountPath = curConfig['server_fuse']['path']
+        self.extraMountPath = tempfile.mkdtemp()
+        self.extraMount = None
+        self.knownPaths = {
+            'user/admin/Private/Item 1/File 1A': 'File 1A',
+            'user/admin/Private/Item 1/File 1B': 'File 1B',
+            'user/admin/Private/Item 2/File 2': 'File 2',
+            'user/admin/Private/Item Without File/': None,
```

Comment:
Some of your logic could be simplified if you split this out into ```self.knownPaths``` and ```self.knownBadPaths```

Reply:
Here, the entry with a None isn't bad in any way.  The parent path exists, but there is no file in this item.  In several tests, we check if the parent filesystem folders (Girder users, collections, folders, or items) exist and have certain properties.  If there is a Girder file, we also check if the filesystem has a file.  Splitting this into two objects will result in duplicating many of the test loops, since we want to check that the parent folders have those properties regardless.

If you think splitting this up and making numerous more tests aids clarity, then I can do that.  I fear it will reduce clarity, since it will no longer be obvious that we are testing for the same behavior of the parent folders.